share/doc/homebrew/How-To-Open-a-Homebrew-Pull-Request-(and-get-it-merged).md

```diff
@@ -35,3 +36,68 @@ To make changes based on feedback:
 Once all feedback has been addressed and if it's a change we want to include (we include most changes) then we'll add your commit to Homebrew. Note it will not show up as "Merged" because of the way we include contributions.
 
 Well done, you are now a Homebrew contributor!
+
+## Staying in Sync
+
+While you have your branch checked out, `brew` won't behave the way you're used to:
+
+* `brew update` will work and report updated formulae, but
+* `brew upgrade` won't upgrade.
+
+**To get formula updates**, switch back to `master` and update as usual:
+
+```sh
+cd $(brew --prefix)
+
+git checkout master
+brew update
+brew upgrade
+```
+
+Your `brew update` and `brew upgrade` will work as expected. Your formula changes won't be available until you rebase: see below.
+
+If `brew update` outputs tips on how to "restore the stashed changes", you had edited some files but not committed the changes. Follow the `git stash pop` advice to recover.
+
+**To rebase your branch** after the steps above, bringing it up to date with `master`:
+
+```sh
+cd $(brew --prefix)
+
+git checkout YOUR_BRANCH_NAME
+git rebase --interactive origin/master
```

Comment:
I think we can just recommend the steps we want people to take inline in the existing text and slim down the rest e.g. `git commit -a; brew update; brew rebase --interactive origin/master; git push --force-with-lease`. While this documentation is good it makes a short document a lot longer.